core/src/main/java/io/undertow/server/protocol/framed/AbstractFramedStreamSinkChannel.java

```diff
@@ -73,8 +73,8 @@ public abstract class AbstractFramedStreamSinkChannel<C extends AbstractFramedCh
 
 
     static {
-        final int defaultAwaitWritableTimeout = 600000;
-        int await_writable_timeout = AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Integer.getInteger("io.undertow.await_writable_timeout", defaultAwaitWritableTimeout));
+        final long defaultAwaitWritableTimeout = TimeUnit.MINUTES.toNanos(10);
+        long await_writable_timeout = AccessController.doPrivileged((PrivilegedAction<Long>) () -> Long.getLong("io.undertow.await_writable_timeout", defaultAwaitWritableTimeout));
```

Comment:
This would break compatibility - users were setting `io.undertow.await_writable_timeout` thinking they are setting milis, but now that value would be taken directly and treated to nanoseconds.

Reply:
Thanks for the answer. I was thinking that maybe I could edit some documentation to mention the change of property for better readability to seconds or minutes. But for now, we can keep the system property in milliseconds and convert it to nanos after reading it.